server/src/main/java/org/opensearch/index/mapper/DerivedFieldType.java

```diff
@@ -159,13 +161,20 @@ public IndexFieldData.Builder fielddataBuilder(String fullyQualifiedIndexName, S
     @Override
     public Query termQuery(Object value, QueryShardContext context) {
         Query query = typeFieldMapper.mappedFieldType.termQuery(value, context);
+        boolean rewrite = true;
+        if (query instanceof IndexOrDocValuesQuery) {
+            if (((IndexOrDocValuesQuery) query).getIndexQuery() instanceof PointRangeQuery) {
+                rewrite = false;
+            }
+        }
```

Comment:
We can abstract this logic into a method and directly have it as part of `DerivedFieldQuery`? instead of passing this rewrite parameter from `DerivedFieldType`?